examples/showcase/firefox.py

```diff
@@ -22,50 +22,38 @@ def svg_parse(path):
                 'Q': (Path.CURVE3,)*2,
                 'C': (Path.CURVE4,)*3,
                 'Z': (Path.CLOSEPOLY,)}
-    path_re = re.compile(r'([MLHVCSQTAZ])([^MLHVCSQTAZ]+)', re.IGNORECASE)
-    float_re = re.compile(r'(?:[\s,]*)([+-]?\d+(?:\.\d+)?)')
     vertices = []
     codes = []
-    last = (0, 0)
-    for cmd, values in path_re.findall(path):
-        points = [float(v) for v in float_re.findall(values)]
-        points = np.array(points).reshape((len(points)//2, 2))
+    cmd_values = re.split("([A-Za-z])", path)[1:]  # Split over commands.
+    for cmd, values in zip(cmd_values[::2], cmd_values[1::2]):
+        # Numbers are separated either by commas, or by +/- signs (but not at
+        # the beginning of the string).
+        points = ([*map(float, re.split(",|(?<!^)(?=[+-])", values))] if values
+                  else [(0., 0.)])  # Only for "z/Z" (CLOSEPOLY).
+        points = np.reshape(points, (-1, 2))
         if cmd.islower():
-            points += last
-        cmd = cmd.capitalize()
-        last = points[-1]
-        codes.extend(commands[cmd])
-        vertices.extend(points.tolist())
-    return codes, vertices
+            points += vertices[-1]
+        codes.extend(commands[cmd.upper()])
+        vertices.extend(points)
+    return np.array(codes), np.array(vertices)
```

Comment:
Probably should append instead?
```suggestion
        vertices.append(points)
    return np.array(codes), np.concatenate(vertices)
```

Reply:
sure